internal/controller/factory/statefulset.go

```diff
@@ -70,6 +71,28 @@ func CreateOrUpdateStatefulSet(
 
 	volumes := generateVolumes(cluster)
 
+	basePodSpec := corev1.PodSpec{
+		Containers: []corev1.Container{generateContainer(cluster)},
+		Volumes:    volumes,
+	}
+	basePodSpecBytes, err := json.Marshal(basePodSpec)
```

Comment:
Looks like merge logic is pretty common and can be reused in multiple places. Maybe better to move to some helper

Reply:
I would like to break this out into a separate function, but once I do, it immediately needs coverage with unit tests. How about we leave it as is and, once (if) we merge, I'll make an issue to factor it out and write some tests?